### PR TITLE
Add workflow to build gem via GH Actions

### PR DESCRIPTION
**Pull request by ashmaroli**
First step towards shipping releases via GH Actions.

### CODE REVIEW

File: .github/workflows/gem.yml [hunk 1]
```diff
@@ -2,9 +2,16 @@ name: Gem Tasks
...
+    branches:
+      - master
+      - gem-tasks-workflow # FIXME: Remove before merge
+      - "*-stable"
+    tags:
+      - "v*"
 
 jobs:
   gem_build:
+    if: "github.repository_owner == 'jekyll' && startsWith(github.event.commits[0].message, 'Release :gem:')"
```
Comment (parkr) [inline .github/workflows/gem.yml:13]: Should we do this when we add a tag, instead? So when we add a tag, the workflow runs and a `.gem` is packaged and released.

File: .github/workflows/gem.yml [hunk 2]
```diff
@@ -20,3 +27,15 @@ jobs:
         with:
           ruby-version: ${{ matrix.ruby_version }}
           bundler-cache: true
+      - name: Extract Plugin Version
+        id: plugin-meta
+        run: echo "::set-output name=version::$(bundle exec ruby -e 'require "jekyll_test_plugin"; puts JekyllTestPlugin::VERSION')"
+        shell: bash
+      - name: "Build Gem (JekyllTestPlugin ${{ steps.plugin-meta.outputs.version }})"
+        run: gem build jekyll_test_plugin.gemspec
```
Comment (parkr) [inline .github/workflows/gem.yml:34]: Eventually `gem push` would be great to add here too since most folks download from RubyGems.org!

**Review by parkr — APPROVED** [review]

File: .github/workflows/gem.yml (hunk 1 above)
Comment (ashmaroli) [reply .github/workflows/gem.yml:13]: > Should we do this when we add a tag, instead?

The workflow is configured to run when a tag is pushed as well.
Regarding whether the clause is valid for `push` event with a tag, that aspect comes into play after this PR gets merged to the default branch.

The idea is to trigger a release when a pushed tag was only created with the message `Release :gem: ...`. (*case-sensitive for enforcing consistency*).

File: .github/workflows/gem.yml (hunk 2 above)
Comment (ashmaroli) [reply .github/workflows/gem.yml:34]: Yes. That's the whole point of using this repository for testing.
The third stage of testing involves checking whether the API Key is accessible via this repo.. 
(*The second stage being centered around tags*).

**Comment by ashmaroli** [comment]
@jekyllbot: merge +dev

**Pull request merged**